sdk/keyvault/keyvault-certificates/review/keyvault-certificates.review.api.md

```diff
@@ -0,0 +1,286 @@
+## API Report File for "@azure/keyvault-certificates"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import * as coreHttp from '@azure/core-http';
+import { HttpClient } from '@azure/core-http';
+import { HttpPipelineLogger } from '@azure/core-http';
+import { PagedAsyncIterableIterator } from '@azure/core-paging';
+import { RequestOptionsBase } from '@azure/core-http';
+import { ServiceClientOptions } from '@azure/core-http';
+import { TokenCredential } from '@azure/core-http';
+
+// @public
+export interface CertificateIssuer {
+    id?: string;
+    provider?: string;
+}
+
+// @public
+export interface CertificateOperation {
+    cancellationRequested?: boolean;
+    csr?: Uint8Array;
+    error?: ErrorModel;
+    readonly id?: string;
+    issuerParameters?: IssuerParameters;
+    requestId?: string;
+    status?: string;
+    statusDetails?: string;
+    target?: string;
+}
+
+// Warning: (ae-forgotten-export) The symbol "CertificateAttributes" needs to be exported by the entry point index.d.ts
+// 
+// @public
+export interface CertificatePolicy extends SecretProperties, CertificateAttributes {
+    certificateTransparency?: boolean;
+    certificateType?: string;
+    ekus?: string[];
+    exportable?: boolean;
+    readonly id?: string;
+    issuerName?: string;
+    // Warning: (ae-forgotten-export) The symbol "JsonWebKeyCurveName" needs to be exported by the entry point index.d.ts
+    keyCurveType?: JsonWebKeyCurveName;
+    keySize?: number;
+    // Warning: (ae-forgotten-export) The symbol "JsonWebKeyType" needs to be exported by the entry point index.d.ts
+    keyType?: JsonWebKeyType;
+    // Warning: (ae-forgotten-export) The symbol "KeyUsageType" needs to be exported by the entry point index.d.ts
+    keyUsage?: KeyUsageType[];
+    lifetimeActions?: LifetimeAction[];
+    reuseKey?: boolean;
+    // Warning: (ae-forgotten-export) The symbol "SubjectAlternativeNames" needs to be exported by the entry point index.d.ts
+    subjectAlternativeNames?: SubjectAlternativeNames;
+    subjectName?: string;
+    validityInMonths?: number;
+}
+
+// @public
+export interface CertificateProperties extends ParsedKeyVaultEntityIdentifier {
+    readonly created?: Date;
+    enabled?: boolean;
+    readonly expires?: Date;
+    readonly id?: string;
```

Comment:
Are all of these fields really optional? For instance, 'id' is probably not?

Reply:
I believe id can be null in this case possibly for pending certificates.

Reply:
perhaps this should be certifcateId on the properties and just id on Certificate will follow up with other languages